### ISSUE TITLE
Splitbutton: display as button in case no children rendered

### ISSUE THREAD

**Issue by Rapster** [primefaces/primefaces#2690]
If there is not child items defined in SplitButton, don't display arrow (e.g should look like a button then). It makes think there is something wrong...

**Comment by tandraschko** [comment]
Thanks for the PR!

**Comment by Rapster** [comment]
I should have put XHTML sample in that issue, my bad... What I meant it, if **no children are rendered**, display `SplitButton` as a simple button

```xml
<p:splitButton value="#{i18n['BUTTON_SAVE']}" action="#{recordController.update}" ajax="false">
    <p:menuitem value="#{i18n['BUTTON_PREFILL']}"
                           rendered="#{recordController.prefillRendered}"
                           actionListener="#{recordController.prefill}">
    </p:menuitem>
</p:splitButton>
```

Based on `ComponentUtils.shouldRenderFacet()`,  we could have `shouldRenderChildren()`

**Comment by Rapster** [comment]
I'll provide a PR

**Comment by VsevolodGolovanov** [comment]
In 6.2.30 splitButton without children causes js errors when clicking anywhere on the page: `Uncaught TypeError: Cannot read property 'left' of undefined`.
Seems to be fixed in 7.0 by #3435.

Can be fixed in 6.2.30 by adding offset check (see `//if there are no menuitems, then the offset is undefined`):
```
PrimeFaces.widget.SplitButton.prototype.bindEvents = function() {
	var $this = this;

	PrimeFaces.skinButton(this.button).skinButton(this.menuButton);

	//mark button and descandants of button as a trigger for a primefaces overlay
	this.button.data('primefaces-overlay-target', true).find('*').data('primefaces-overlay-target', true);

	//toggle menu
	this.menuButton.click(function() {
		if ($this.menu.is(':hidden'))
			$this.show();
		else
			$this.hide();
	});

	//menuitem visuals
	this.menuitems.mouseover(function(e) {
		var menuitem = $(this),
			menuitemLink = menuitem.children('.ui-menuitem-link');

		if (!menuitemLink.hasClass('ui-state-disabled')) {
			menuitem.addClass('ui-state-hover');
		}
	}).mouseout(function(e) {
		$(this).removeClass('ui-state-hover');
	}).click(function() {
		$this.hide();
	});

	//keyboard support
	this.menuButton.keydown(function(e) {
		var keyCode = $.ui.keyCode;

		switch (e.which) {
			case keyCode.UP:
				$this.highlightPrev(e);
				break;

			case keyCode.DOWN:
				$this.highlightNext(e);
				break;

			case keyCode.ENTER:
			case keyCode.NUMPAD_ENTER:
			case keyCode.SPACE:
				$this.handleEnterKey(e);
				break;


			case keyCode.ESCAPE:
			case keyCode.TAB:
				$this.handleEscapeKey();
				break;
		}
	});

	var hideNS = 'mousedown.' + this.id;
	$(document.body).off(hideNS).on(hideNS, function(e) {
		//do nothing if hidden already
		if ($this.menu.is(":hidden")) {
			return;
		}

		//hide overlay if mouse is outside of overlay except button
		var offset = $this.menu.offset();

		//if there are no menuitems, then the offset is undefined
		if (!offset) {
			return;
		}

		if (e.pageX < offset.left ||
			e.pageX > offset.left + $this.menu.width() ||
			e.pageY < offset.top ||
			e.pageY > offset.top + $this.menu.height()) {

			$this.button.removeClass('ui-state-focus ui-state-hover');
			$this.hide();
		}
	});

	var resizeNS = 'resize.' + this.id;
	$(window).off(resizeNS).on(resizeNS, function() {
		if ($this.menu.is(':visible')) {
			$this.alignPanel();
		}
	});

	if (this.cfg.filter) {
		this.setupFilterMatcher();
		this.filterInput = this.menu.find('> div.ui-splitbuttonmenu-filter-container > input.ui-splitbuttonmenu-filter');
		PrimeFaces.skinInput(this.filterInput);

		this.bindFilterEvents();
	}
};
```

Reproducer:
```
	<h:body>
		<style>
			html, body, form {
				height: 100%;
			}
		</style>
		<h:form>
			<p:splitButton value="test" />
		</h:form>
	</h:body>
```
